tamara_gadieva/graph.cpp

```diff
@@ -42,34 +38,73 @@ void Vertex::add_edge_id(const EdgeId& id) {
   edge_ids_.push_back(id);
 }
 
-VertexId Graph::add_vertex(const int depth) {
+void Graph::fill_vertices_ids_in_depth(int depth, VertexId id) {
+  if (depth >= vertices_ids_in_depth_.size())
+    vertices_ids_in_depth_.resize(depth + 1);
```

Comment:
1. Не логично, что ты проверяешь просто больше ли, а добавляешь всего 1 элемент.
    1) Либо проверяй что они равны и нам нужна новая глубина
    2) Либо добавляй количество элементов, равное разнице
2. У тебя есть интерфейс для получения глубины, вот его и используй
3. `resize()` работает, но такая логика больше сишная. Ты что хочешь сделать тут? Добавить новый элемент в вектор, вот и добавь новый элемент
```suggestion
  assert(depth <= get_depth());
  if (depth == get_depth())
    vertices_ids_in_depth_.emplace_back();
```